Blockzilla/SettingsViewController.swift

```diff
@@ -8,22 +8,66 @@ import UIKit
 import Telemetry
 
 class SettingsTableViewSearchCell: UITableViewCell {
-    let accessoryLabel = UILabel()
+    private let newLabel = UILabel()
+    private let accessoryLabel = UILabel()
+    private let spacerView = UIView()
+
+    var accessoryLabelText: String? {
+        get { return accessoryLabel.text }
+        set {
+            accessoryLabel.text = newValue
+            accessoryLabel.sizeToFit()
+        }
+    }
+
+    var label: String? {
+        get { return newLabel.text }
+        set {
+            newLabel.text = newValue
+            newLabel.sizeToFit()
+        }
+    }
 
     override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
         super.init(style: style, reuseIdentifier: reuseIdentifier)
+        newLabel.numberOfLines = 0
+        newLabel.lineBreakMode = .byWordWrapping
+        textLabel?.numberOfLines = 0
+        textLabel?.text = " "
+
         contentView.addSubview(accessoryLabel)
+        contentView.addSubview(newLabel)
+        contentView.addSubview(spacerView)
+
+        newLabel.textColor = UIConstants.colors.settingsTextLabel
         accessoryLabel.textColor = UIConstants.colors.settingsDetailLabel
         accessoryType = .disclosureIndicator
 
         let backgroundColorView = UIView()
         backgroundColorView.backgroundColor = UIConstants.colors.cellSelected
         selectedBackgroundView = backgroundColorView
 
+        accessoryLabel.setContentHuggingPriority(.required, for: .horizontal)
+        accessoryLabel.setContentCompressionResistancePriority(.required, for: .horizontal)
         accessoryLabel.snp.makeConstraints { make in
-            make.centerY.equalToSuperview()
-            make.trailing.equalToSuperview()
+            make.trailing.centerY.equalToSuperview()
         }
+
+        newLabel.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+        newLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+
+        spacerView.snp.makeConstraints { make in
```

Comment:
What's the point of the spacerView?

Reply:
For the auto-resizing tableViewCell to work I need views anchored to all 4 edges of the contentView. But the newLabel can't have it's leading attached to the contentView AND the textLabel. So the spacerView is there to fill that gap